kv_backend.cc

```diff
@@ -1,33 +1,71 @@
-#include <string>
-#include <map>
-#include "key_value_interface.cc"
-
-// true kv_backend implementation that stores all raw data
-class KeyValueBackEnd : public KeyValueInterface{
-	public:
-		KeyValueBackEnd();
-		~KeyValueBackEnd();
-		std::string Get(const std::string& key);
-		std::string Put(const std::string& key, const std::string& chirp_value);
-		std::string DeleteKey(const std::string& key);
-	private:
-		// key - std::string, value - serialized chirp or user object
-		std::map<std::string, std::string> chirp_map_;	
-};
-
-std::string KeyValueBackEnd::Get(const std::string& key) {
-	return chirp_map_.at(key);
+#include "kv_backend.h"
+
+#include <iostream>
+
+// global mutex that keeps the database thread safe
+std::mutex mutex_lock;
+
+KeyValueBackEnd::KeyValueBackEnd() {
+	// initialize the key counter to 0 when the database is created
```

Comment:
I personally think this kind of logic should be in the service layer bc what if someone else wants to use your kv back end with their own service layer? And they don't use the same naming? The kv backend shouldnt know anything about what it is storing